api/app/helpers/seeds_handler.rb

```diff
@@ -0,0 +1,184 @@
+            return JSON.parse(mock_session.last_response.body, symbolize_names: true)
+        rescue JSON::ParserError
+            error = "#{action.to_s.upcase} #{path} is an invalid path."
+            return {status: 'error', message: error, payload: {}}
+        end
+    end
+
+    def read_json(file, log)
+        seed_data_dir = Rails.root.join('db', 'seed')
+        file_dir = seed_data_dir+file
+        begin
+            data = File.read(file_dir)
+            return JSON.parse(data, symbolize_names: true), log
+        rescue Errno::ENOENT => e
+            log.push(e)
+            return [], log
+        rescue JSON::ParserError 
+            log.push("#{file_dir} is not a valid JSON file.")
+            return [], log
+        end
+    end
+
+    def insert_data(chaining, file)
+        log, records = [], {}
+        chaining, records = get_existing_records(chaining, records)
+        if file
+            json, log = read_json(file, log)
+            records = set_available_position_templates(json, records)
+            log = post_data(chaining, json, records, log)
+        else
+            log.push("No JSON given.")
+        end
+        puts log
+    end
+
+    def post_data(chaining, json, records, log)
+        chaining.each do |entry|
+            key = entry[:label]
+            table_data = []
+            errors = []
+            json[key].each do |data|
+                data = replace_all_index(json, data)
+                route = get_route(entry[:create], data)
+                res = request(:post, route, data)
+                if res[:status] == 'success'
+                    if array_result(route)
+                        table_data.push(get_entry(data, res[:payload], entry[:unique]))
+                    else
+                        table_data.push(res[:payload])
+                    end
+                else
+                    errors.push(res[:message])
+                    table_data.push(nil)
+                end
+            end
+            json[key] = table_data
+            if len(errors) == 0
+                log.push("Insertions/updates to the table #{key} were successful.")
+            else
+                log.push("There were errors with the insertion/update to table #{key}.")
+                log += errors
+            end
+        end
+        return log
+    end
+
+    def get_entry(data, result, unique)
+        result.each do |entry|
+            if match_all_unique_keys(data, entry, unique)
+                return entry
+            end
+        end
+    end
+
+    def match_all_unique_keys(data, entry, unique)
+        unique.each do |key|
+            if is_index(key)
+                key = id_name(key)
+            end
+            if data[key] != entry[key]
+                return false
+            end
+        end
+        return true
+    end
+
+    def array_result(route)
+        return route.scan(/(\/add_)/).length > 0
+    end
+
+    def get_route(route, data)
```

Comment:
can we add the name of the route into chaining? This avoids potential parsing errors.

Reply:
Sure.

Reply:
Actually, I can't because I don't know the value of `:session_id` or `:id`, so I can't include them.